src/Serializer/AbstractItemNormalizer.php

```diff
@@ -171,19 +171,54 @@ public function supportsDenormalization($data, $type, $format = null, array $con
      */
     public function denormalize($data, $class, $format = null, array $context = [])
     {
+        $resourceClass = $this->resourceClassResolver->getResourceClass(null, $class);
         $context['api_denormalize'] = true;
-        $context['resource_class'] = $class;
-        $inputClass = $this->getInputClass($class, $context);
+        $context['resource_class'] = $resourceClass;
 
-        if (null !== $inputClass && null !== $dataTransformer = $this->getDataTransformer($data, $class, $context)) {
-            return $dataTransformer->transform(
-                parent::denormalize($data, $inputClass, $format, ['resource_class' => $inputClass] + $context),
-                $class,
-                $context
-            );
+        if (null !== ($inputClass = $this->getInputClass($resourceClass, $context)) && null !== ($dataTransformer = $this->getDataTransformer($data, $resourceClass, $context))) {
+            $dataTransformerContext = $context;
+
+            unset($context['input']);
+            unset($context['resource_class']);
```

Comment:
Could be merged with the previous call.

Reply:
But it'd be harder to read... :disappointed:

Reply:
Looks like there's absolutely no difference in terms of opcodes:

https://3v4l.org/GREph/vld#output

vs

https://3v4l.org/3kNUI/vld#output